### ISSUE TITLE
Onnx - Consider as way to share models

### ISSUE THREAD

**Issue by cjerzak** [cjerzak/LinkIt-software#2]
https://en.wikipedia.org/wiki/Open_Neural_Network_Exchange

https://huggingface.co/blog/convert-transformers-to-onnx

**Comment by beniaminogreen** [comment]
Hi Connor, 

Did you manage to look into this? I am thinking of experimenting with [this rust wrapper for the ONNX runtime](https://github.com/pykeio/ort) as a way to provide fast inference in R for ML models without needing a python install.  If we are able to export the models you have built using the ONNX format, we might be able to quickly perform inference using this approach.

Best, 
Ben

**Comment by cjerzak** [comment]
true. i haven't taken a close look but will do so later this/next week! we do want to simplify the inference pipeline...

**Comment by beniaminogreen** [comment]
Just got back from travels and after a long time working out the compilation + some dependency steps, I am excited to report that I am able to deploy GPT-2 on my laptop using this approach. 

Seems like this could be a very promising avenue for deploying our models if they are easy to translate into the appropriate format. Ideally, we could pre-process everything in R, then pass the data to the ONNX runtime to do the difficult steps efficiently. Will provide updates as I make more progress.

**Comment by cjerzak** [comment]
very cool! yes, minimizing dependencies for people will be very
important... happy to talk anytime this week.
